xtext-gradle-plugin/src/main/java/org/xtext/gradle/XtextBuilderPlugin.xtend

```diff
@@ -162,14 +164,15 @@ class XtextBuilderPlugin implements Plugin<Project> {
 							javaCompile.dependsOn(generatorTask)
 							javaCompile.doLast(new Action<Task>() {
 								override void execute(Task it) {
-									generatorTask.installDebugInfo()
+									if (it instanceof AbstractCompile) {
```

Comment:
This is always true, you are configuring a JavaCompile task here.

Reply:
in the closure i have a task only

Reply:
=> how shall it work without instanceof / downcast

Reply:
javacompiler returns a task thus i need an action<task>

Reply:
The javaCompile variable is already of type JavaCompile

Reply:
ahh this is different at android here for some reason

Reply:
but

```
javaCompile.doLast(new Action<JavaCompile>() {
								override void execute(JavaCompile it) {
									if (it instanceof AbstractCompile) {
										generatorTask.installDebugInfo(destinationDir)
									}
								}
							})
``` does not compile

Reply:
so you want to capture the original javacompile here?

Reply:
Yeah why not? It's the same instance.

Reply:
```
override void execute(Task it) {
									generatorTask.installDebugInfo(javaCompile.destinationDir)
								}
``

Reply:
hmmm
so you think https://github.com/xtext/xtext-gradle-plugin/compare/cd_issue154new2?expand=1
is better?

Reply:
Yes, but only for this one. You can't cast the Android plugins task to JavaCompile because they have another task that can be used instead (although almost no one uses it anymore). So there we need to keep the check.

Reply:
@oehme i updated the pr for here